velox/type/LongDecimal.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <folly/dynamic.h>
+#include <sstream>
+#include <string>
+#include "velox/common/base/BitUtil.h"
+#include "velox/common/base/Exceptions.h"
+#include "velox/type/StringView.h"
+
+#pragma once
+
+#define BUILD_INT128(X, Y) ((int128_t)(X) << 64 | Y)
+
+namespace facebook::velox {
+
+using int128_t = __int128_t;
+
+struct LongDecimal {
+ public:
+  // Default required for creating vector with NULL values.
+  constexpr explicit LongDecimal() : unscaledValue_(0) {}
```

Comment:
Do we really need to zero initializing the data?  This could impact performance.

Reply:
Isn't a constexpr evaluated at compile time? so wherever the default constructor is called, unscaledValue_(0) is already initialized right.

Reply:
I mean if you have something like `new LongDecimal[1'000'000]`, it needs to call `bzero` or `memzero` on all 16 MB memory, which is not always necessary.

Reply:
If we don't initialize the unscaledValue_ in the default constructor it cannot be constexpr anymore. To make a ctor constexpr, it must initialize its members.

Reply:
Why the default constructor needs to be `constexpr`?  Where is it used?

Reply:
> Why the default constructor needs to be constexpr? Where is it used?
 Like mentioned in the comment, the default constructor is used while creating vector with NULL values.

> If we can make sure this is POD like (mentioned in another comment), we are probably safe for buffer allocation. But still need to be careful about adhoc allocation.
is_pod fails only if there are copy constructors.